include/mqt-core/dd/Node.hpp

```diff
@@ -40,12 +37,23 @@ struct NodeBase : LLBase {
    * @details Not required for all node types, but padding is required either
    * way.
    *
-   * 0b1000 = marks a reduced dm node,
-   *  0b100 = marks a dm (tmp flag),
-   *   0b10 = mark first path edge (tmp flag),
-   *    0b1 = mark path is conjugated (tmp flag)
+   * 0b10000 = mark flag used for mark-and-sweep garbage collection,
+   *  0b1000 = marks a reduced dm node,
+   *   0b100 = marks a dm (tmp flag),
+   *    0b10 = mark first path edge (tmp flag),
+   *     0b1 = mark path is conjugated (tmp flag)
    */
-  std::uint16_t flags = 0;
+  std::uint16_t flags = 0; // TODO: Would it make sense to use a larger datatype
+                           // here since Ref is gone?
```

Comment:
Yeah. I was thinking about that already.
Initially, I would have hoped that we get some space savings for the nodes here, but that turned out to be wishful thinking. So we might as well make the padding explicit.
However that's also kind of awkward as there is no 48bit type.

Reply:
What's your take on using a [bit field](https://en.cppreference.com/w/cpp/language/bit_field.html)? All I know is that they exist.

Reply:
Hm. Just read through the linked page.
Sounds reasonable in principle. I am a bit worried that the standard text mentions the implementation defined nature of the allocation details quite often.
We should, at least, make sure that on the platforms that we commonly test with, the packing and alignment of the resulting struct is as we would expect it.

Reply:
That looks really reasonable. and convenient.
Seems to be something for a separate PR, but I definitely like the idea! 👍🏼